k8s/dialer_test.go

```diff
@@ -35,7 +37,7 @@ func TestDialInClusterService(t *testing.T) {
 
 	testingNS := &coreV1.Namespace{
 		ObjectMeta: metaV1.ObjectMeta{
-			Name: "dialer-test-ns",
+			Name: "dialer-test-ns-" + rand.String(5),
```

Comment:
This looks good but I think maybe a `t.Cleanup` removing the namespace would be a better solution for people running the e2e suite locally otherwise they may end up with a bunch of `dialer-test-ns-***` namespaces without probably noticing